lib/ansible/modules/cloud/vmware/vmware_content_deploy_template.py

```diff
@@ -0,0 +1,273 @@
+        self.host_id = self.get_host_by_name(self.content_service, self.datacenter, self.host)
+        if not self.host_id:
+            self.module.fail_json(msg="Failed to find the Host %s" % self.host)
+        # Find the resourcepool by the given resourcepool name
+        self.resourcepool_id = None
+        if self.resourcepool:
+            self.resourcepool_id = self.get_resource_pool_by_name(self.content_service, self.datacenter, self.resourcepool)
+            if not self.resourcepool_id:
+                self.module.fail_json(msg="Failed to find the resource_pool %s" % self.resourcepool)
+        # Find the Cluster by the given Cluster name
+        self.cluster_id = None
+        if self.cluster:
+            self.cluster_id = self.get_resource_pool_by_name(self.content_service, self.datacenter, self.resourcepool)
+            if not self.cluster_id:
+                self.module.fail_json(msg="Failed to find the Cluster %s" % self.cluster)
+        # Create VM placement specs
+        self.placement_spec = LibraryItems.DeployPlacementSpec(folder=self.folder_id,
+                                                               host=self.host_id
+                                                               )
+        if self.resourcepool_id or self.cluster_id:
+            self.placement_spec.resource_pool = self.resourcepool_id
+            self.placement_spec.cluster = self.cluster_id
+        self.vm_home_storage_spec = LibraryItems.DeploySpecVmHomeStorage(datastore=to_native(self.datastore_id))
+        self.disk_storage_spec = LibraryItems.DeploySpecDiskStorage(datastore=to_native(self.datastore_id))
+        self.deploy_spec = LibraryItems.DeploySpec(name=self.vm_name,
+                                                   placement=self.placement_spec,
+                                                   vm_home_storage=self.vm_home_storage_spec,
+                                                   disk_storage=self.disk_storage_spec,
+                                                   powered_on=power_on
+                                                   )
+        vm_id = self.template_service.deploy(self.library_item_id, self.deploy_spec)
+        if vm_id:
+            self.module.exit_json(
+                changed=True,
+                vm_deploy_info=dict(
+                    msg="Deployed Virtual Machine '%s'." % self.vm_name,
+                    vm_id=vm_id,
+                )
+            )
+        self.module.exit_json(changed=False,
+                              vm_deploy_info=dict(msg="Virtual Machine deployment failed", vm_id=''))
+
+
+def main():
+    argument_spec = VmwareRestClient.vmware_client_argument_spec()
+    argument_spec.update(
+        state=dict(type='str', default='present',
+                   choices=['present', 'poweredon']),
+        template=dict(type='str', aliases=['template_src'], required=True),
+        name=dict(type='str', required=True, aliases=['vm_name']),
```

Comment:
There are other parameter to find an existing VM using `get_vm` API such as `moid`, `uuid` and a helper parameter `use_instance_uuid`, it would be great if we can implement these parameters as well. This will make all guest module parameters equal and this module future proof.